cpp/src/arrow/filesystem/localfs.cc

```diff
@@ -210,6 +211,22 @@ Result<FileInfo> StatFile(const std::string& path) {
   return info;
 }
 
+Result<FileInfo> IdentifyFile(const std::string& path) {
+  FileInfo info;
+
+  if (std::filesystem::is_directory(path)) {
+    info.set_type(FileType::Directory);
+  } else if (std::filesystem::is_regular_file(path)) {
+    info.set_type(FileType::File);
+  }
```

Comment:
`std::filesystem` supports a bunch of file types: https://en.cppreference.com/w/cpp/filesystem. Should we add any of the others to `else if`? Add a case for an empty path?

Reply:
We should probably ignore entries that aren't files or directories. Except maybe symlinks.